src/client.rs

```diff
@@ -73,6 +88,15 @@ impl Client {
         }
     }
 
+    pub fn discard_greeting(&mut self) {
```

Comment:
Hm... I'm not sure I understand this. Do we need this? I'm asking because we already set the state via the `Options`? Also, as this merhod is `pub`, it may be called at unusual times and interfer with our state? Maybe I'm wrong.

Reply:
I thought it could be useful in some cases. For example, when a Client is already initialized and STARTTLS is prompted later on (at runtime): it would be handy to just discard greeting rather than re-creating a new Client.

It does not cost that much to have setters, and it's a question of time someone will need it for a specific flow. Plus chances to break everything is low (as it needs to be done before receiving greetings). We could add a condition not to reset the state in case greeting is already discarded? Let me push a safer version.

That said, I would be fine not to add them as well, it was just a proposition. Let me know.